### PR TITLE
Overlay fullfield image / heat map updates

### PR DESCRIPTION
**Pull request by kimberlyh66**
Updates made to:
- overlay fullfield image (addresses issue #10)
- add toggle button to display heat map 
- speed up rendering of heat map 
- add scan options table / toggle buttons for corresponding scans only if thumbs exist for selected date

### CODE REVIEW

**Comment by dlebauer** [comment]
When I try to run this (both on workbench and on my laptop) I get the following error:

> Error in as.Date.default(unique(unlist(str_extract_all(list.files(image_dir),  : 
  do not know how to convert 'unique(unlist(str_extract_all(list.files(image_dir), "[0-9]{4}-[0-9]{2}-[0-9]{2}")))' to class “Date”

File: app.R [hunk 1]
```diff
@@ -30,6 +32,12 @@ load_cache <- function(full_cache_data) {
     }
 }
 
+# directory containing full-field images
+image_dir <- 'data/terraref/sites/ua-mac/Level_2/rgb_fullfield/_thumbs'
```
Comment (dlebauer) [inline app.R]:
```suggestion
image_dir <- '~/data/terraref/sites/ua-mac/Level_2/rgb_fullfield/_thumbs'
```

**Comment by Chris-Schnaufer** [comment]
Updating with David's requested changes

**Comment by kimberlyh66** [comment]
@dlebauer suggestions have been addressed.

I have made some notes in the README on where to download the thumbs (`/ua-mac/Level_2/rgb_fullfield/_thumbs/` endpoint on Globus) and where to save on local computer (`~/data/terraref/sites/ua-mac/Level_2/rgb_fullfield/_thumbs`).

Since I do not have this path on my computer `cache_path <- "/srv/shiny-server/cache/cache.RData"`, I usually comment out lines 14-33, 377 and add the following lines to replace line 377 to get the app to run:

```r
load('cache_full.RData')
full_cache_data <- full_cache_data["MAC Season 6"] 
```
cache_full.RData is a replacement for the cache file and should be saved in the working directory.

**Comment by kimberlyh66** [comment]
@dlebauer I usually save a copy of the cache.RData file that you provided a few months ago to the working directory. Do you still have this file? I am unable to attach a .RData file here.

**Comment by dlebauer** [comment]
> @dlebauer I usually save a copy of the cache.RData file that you provided a few months ago to the working directory. Do you still have this file? I am unable to attach a .RData file here.

The cache-refresh script should be able to (and was previously able to) handle the case and create a new cache.RData when the file does not exist. I'm not sure where the feature was lost, but could you please re-implement it?

**Comment by kimberlyh66** [comment]
Sure

**Comment by kimberlyh66** [comment]
I think the cache-refresh.R script is executed when running the app with docker, so it doesn't need to be re-implemented. 

I haven't been able to connect to the database, so I haven't been able to run this script. How do you usually connect to the database?

**Comment by dlebauer** [comment]
I usually use an ssh tunnel that you wouldn’t have access to. But won’t a local docker container work?

________________________________

**Comment by dlebauer** [comment]
still need to resolve a few remaining bugs w/ PR

**Comment by kimberlyh66** [comment]
@dlebauer I have addressed the remaining bugs occurring with this PR. The cache-refresh.R script should work now and you should be able to run app.R without any problems.

I noticed that the rgb thumbs appear to be slightly different in color from the original when displayed on the rendered map. I get the original color when plotting a RasterBrick with `raster::plotRGB`. However, when I try adding the same RasterBrick object to the leaflet map using `mapview::viewRGB`, the color is altered. Do you have any idea why this may be occurring?

**Comment by dlebauer** [comment]
Can you post screenshots of the two? Could it be the order of RGB layers specified by the r,g,b arguments to viewRGB? Or the quantiles argument?

________________________________

**Comment by dlebauer** [comment]
The defaults for r,g,b are reversed in these two functions

A9713CD0-25B6-492F-8C72-90A8AC7BD7D5
294DBEC3-0F5F-44AF-A39E-781C23C7F0E1

**Comment by kimberlyh66** [comment]
original
original

**Comment by kimberlyh66** [comment]
rendered on map
overlay_image

**Comment by dlebauer** [comment]
Does it seem plausible that these have the red and blue channels switched?

**Comment by kimberlyh66** [comment]
Do you mean trying to set r = 1,  g = 2, and b = 3?

I tried running the following code chunk, but still get the same results:

```r
map <- viewRGB(x = fullfield_image,
                       r = 1,
                       g = 2, 
                       b = 3,
                       map = map, layer.name = scan_name)
```

**Comment by dlebauer** [comment]
It is odd it gives the same results ... perhaps try the reverse or moving these up to see if the values affect the output? Also, what is the dimension of the rasterstackbrick?

**Comment by kimberlyh66** [comment]
Dimensions of RasterBrick is `dimensions  : 1285, 164, 210740, 3  (nrow, ncol, ncell, nlayers)`

**Comment by dlebauer** [comment]
looking into it now ... but also getting deprecation warnings like:

> Warning: 'mapview::removeHomeButton' is deprecated.
> Use 'leafem::removeHomeButton' instead.
> See help("Deprecated") and help("mapview-deprecated").
> Warning: 'mapview::addHomeButton' is deprecated.
> Use 'leafem::addHomeButton' instead.

**Comment by dlebauer** [comment]
I've created an issue in the mapview repository https://github.com/r-spatial/mapview/issues/234 to hopefully resolve the color issue and have marked comments above as 'outdated' to shorten the thread a bit.

**Pull request merged**